### ISSUE TITLE
Dim -3 reduce mean unsupported.

### ISSUE THREAD

**Issue by LPanosTT** [tenstorrent/tt-metal#12845]
**Describe the bug**
Dim -3 cannot be reduced on. Throwing `Unsupported dim` at 
```
ttnn/cpp/ttnn/operations/reduction/generic/generic_reductions.cpp:140
```

**To Reproduce**
Run the following test:
```
def test_z_dim_mean(device):
    torch.manual_seed(0)

    input_shape = (1, 7, 7, 2048)
    dim = -3

    torch_input_tensor = torch_random(input_shape, -1, 1, dtype=torch.bfloat16)
    torch_output_tensor = torch.mean(torch_input_tensor, dim=dim, keepdim=True, dtype=torch.bfloat16)

    input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)

    output_tensor = ttnn.mean(input_tensor, dim=dim)
    output_tensor = ttnn.to_torch(output_tensor)
    assert_with_pcc(torch_output_tensor, output_tensor)
```
**Expected behavior**
It reduces along the z-dim

**Additional context**
Blocking resnet50 through the forge --> MLIR flow

**ADDITIONAL REQUEST**
I notice that there is a dual reduce option for YX in ttnn. If when bringing this up, we could get one for ZY (that being dims -2 and -3) as well that would be great.

CC: @nvukobratTT 

**NOTE:** I'm not sure if this should be classified as a bug report, or a feature request. If it should be a feature request feel free to edit the title and label(s)

**Comment by LPanosTT** [comment]
> The conditional statement explicitly does not allow this set up. I think this is a feature request and changed the label/type.
> 
> Could you please elaborate on "I notice that there is a dual reduce option for YX in ttnn."? Preferably with specific examples with hypothetical code.

I mean that as in these are the dimension options for reduce in: `ttnn/cpp/ttnn/operations/reduction/generic/device/common.hpp`
```
enum class ReduceOpDim {
    H, W, HW
};
```

**Comment by bbradelTT** [comment]
@LPanosTT I just rebased on newest main, and things seem to work. I wonder if the negative indexing fix unblocked this scenario.

Could you please try your test on the newest tt-metal main?

**Comment by LPanosTT** [comment]
@bbradelTT i can confirm that this does work as per some MLIR tests that use transposes with -3
